core/lib/rom/schema.rb

```diff
@@ -136,14 +136,26 @@ def self.define(name, attributes: EMPTY_ARRAY, attr_class: Attribute, **options)
       ) { |schema| yield(schema) if block_given? }
     end
 
+    # Builds a representation of the information needed to create an
+    # attribute.
+    #
+    # This representation is consumed by `Schema.define` in order to create
+    # the actual attributes.
+    #
+    # @return [Hash] A hash with `:type` and `:options` keys.
+    #
+    # @api private
+    def self.build_attribute_info(type, options)
```

Comment:
Defining this method here, because it seems that the attribute class is decoupled from the attribute representation (`Schema.define` accepts the attribute class as argument). Otherwise, maybe we could work with `Attribute#ast`, which would be neater.